# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/WebrtcTCPSocketWrapper.cpp
# Commit: b879a0044cf3
# Full Hash: b879a0044cf39f0230879d075c915485affb7707
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Teach WebrtcTCPSocket to bind to a specific host/port. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45102
# ==============================================================================

diff -r f5353ea9f1df -r b879a0044cf3 media/mtransport/WebrtcTCPSocketWrapper.cpp
--- a/media/mtransport/WebrtcTCPSocketWrapper.cpp	Tue Sep 17 18:17:14 2019 +0000
+++ b/media/mtransport/WebrtcTCPSocketWrapper.cpp	Tue Sep 17 18:17:18 2019 +0000
@@ -43,22 +43,24 @@
 }
 
 void WebrtcTCPSocketWrapper::AsyncOpen(
-    const nsCString& aHost, const int& aPort,
-    const shared_ptr<NrSocketProxyConfig>& aConfig) {
+    const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
+    const int& aLocalPort, const shared_ptr<NrSocketProxyConfig>& aConfig) {
   if (!NS_IsMainThread()) {
     MOZ_ALWAYS_SUCCEEDS(mMainThread->Dispatch(
-        NewRunnableMethod<const nsCString, const int,
-                          const shared_ptr<NrSocketProxyConfig>>(
+        NewRunnableMethod<const nsCString, const int, const nsCString,
+                          const int, const shared_ptr<NrSocketProxyConfig>>(
             "WebrtcTCPSocketWrapper::AsyncOpen", this,
-            &WebrtcTCPSocketWrapper::AsyncOpen, aHost, aPort, aConfig)));
+            &WebrtcTCPSocketWrapper::AsyncOpen, aHost, aPort, aLocalAddress,
+            aLocalPort, aConfig)));
     return;
   }
 
   MOZ_ASSERT(!mWebrtcTCPSocket, "wrapper already open");
   mWebrtcTCPSocket = new WebrtcTCPSocketChild(this);
-  mWebrtcTCPSocket->AsyncOpen(
-      aHost, aPort, aConfig->GetLoadInfoArgs(), aConfig->GetAlpn(),
-      dom::TabId(aConfig->GetTabId()), aConfig->GetProxyPolicy());
+  mWebrtcTCPSocket->AsyncOpen(aHost, aPort, aLocalAddress, aLocalPort,
+                              aConfig->GetLoadInfoArgs(), aConfig->GetAlpn(),
+                              dom::TabId(aConfig->GetTabId()),
+                              aConfig->GetProxyPolicy());
 }
 
 void WebrtcTCPSocketWrapper::SendWrite(nsTArray<uint8_t>&& aReadData) {